### PR TITLE
Deployment to AWS using AWS Copilot

### PR DESCRIPTION
**Pull request by EugeneLightsOn**
- AWS deployment using Copilot

**AI Description**

This PR introduces a new script that deploys Cohere's Toolkit to AWS using AWS Copilot. The script creates a new application, environment, and services in AWS Copilot, leveraging containerised DB services for the database. Users can adjust the deployment to use RDS if needed, following the AWS Copilot documentation.

The PR also includes Dockerfiles for various services, such as `aws-api.Dockerfile`, `aws-db.Dockerfile`, `aws-fe.Dockerfile`, `aws-nginx.Dockerfile`, and `aws-terrarium.Dockerfile`. These Dockerfiles define the base images, environment variables, and build instructions for each service.

Additionally, the PR adds manifest files for each service, specifying configuration details like the Docker build arguments, exposed ports, CPU and memory allocations, and environment variables.

The following changes are made:

- A new script is added to deploy Cohere's Toolkit to AWS using AWS Copilot.
- Dockerfiles are introduced for various services, including `aws-api.Dockerfile`, `aws-db.Dockerfile`, `aws-fe.Dockerfile`, `aws-nginx.Dockerfile`, and `aws-terrarium.Dockerfile`.
- Manifest files are added for each service, providing configuration details.
- The `src/backend/alembic/env.py` and `src/backend/database_models/database.py` files are updated to handle database connections when deployed to AWS ECS.

### CODE REVIEW

File: aws_copilot_deploy/aws-api.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
+FROM python:3.11
```
Comment (tianjing-li) [inline aws_copilot_deploy/aws-api.Dockerfile:1]: Thank you for putting all these in a separate folder, great work

Comment (EugeneLightsOn) [reply aws_copilot_deploy/aws-api.Dockerfile:1]: Thanks

File: src/backend/alembic/env.py [hunk 2]
```diff
@@ -9,14 +10,21 @@
...
 load_dotenv()
 
 # Alembic Config object, which provides access to the values inside the alembic.ini file
 config = context.config
 
 # Overwrite alembic.file `sqlachemy.url` value
-config.set_main_option("sqlalchemy.url", Settings().database.url)
+db_url = Settings().database.url
+# if deployed to AWS ECS, use the service discovery endpoint for the database
+aws_discovery_endpoint = os.environ.get("COPILOT_SERVICE_DISCOVERY_ENDPOINT", None)
```
[the remark this answers is not in the document]
Comment (EugeneLightsOn) [reply src/backend/alembic/env.py]: Thanks

File: src/backend/alembic/env.py [hunk 3]
```diff
@@ -9,14 +10,21 @@
...
+if aws_discovery_endpoint:
```
[the remark this answers is not in the document]
Comment (EugeneLightsOn) [reply src/backend/alembic/env.py]: Thank you, TJ

**Review by tianjing-li — APPROVED** [review]

**Pull request merged**